application/blueprints/datamanager/services/async_api.py

```
@@ -88,19 +95,108 @@ def fetch_request(request_id: str) -> dict:
+    wanted = max(total_available - start_offset, 0)
+    if max_rows is not None:
+        wanted = min(wanted, max_rows)
+
+    all_details = list(first_batch)
+    offsets = range(start_offset + len(first_batch), start_offset + wanted, limit)
+    if offsets and first_batch:
+        urls = [
+            f"{url}?{urlencode({'offset': offset, 'limit': limit})}"
+            for offset in offsets
+        ]
+        for offset, page in zip(offsets, fetch_pages_concurrently(urls)):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)